examples/all-clusters-app/all-clusters-common/src/WhmManufacturer.cpp

```diff
@@ -0,0 +1,286 @@
+ *    limitations under the License.
+ */
+
+#include <WhmInstance.h>
+#include <WhmManufacturer.h>
+#include <water-heater-mode.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/water-heater-management-server/WaterHeaterManagementTestEventTriggerHandler.h>
+#include <app/clusters/water-heater-management-server/water-heater-management-server.h>
+
+using namespace chip;
+using namespace chip::app::Clusters::WaterHeaterManagement;
+
+using Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WaterHeaterManagement {
+
+CHIP_ERROR WhmManufacturer::Init()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmManufacturer()->GetWhmDelegate();
+    if (dg == nullptr)
+    {
+        ChipLogError(AppServer, "WhmDelegate is not initialized");
+        return CHIP_ERROR_UNINITIALIZED;
+    }
+
+    dg->SetHeaterTypes(BitMask<WaterHeaterTypeBitmap>(WaterHeaterTypeBitmap::kImmersionElement1));
+    dg->SetHeatDemand(BitMask<WaterHeaterDemandBitmap>(WaterHeaterDemandBitmap::kImmersionElement1));
+    dg->SetEstimatedHeatRequired(10000);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR WhmManufacturer::Shutdown()
+{
+    return CHIP_NO_ERROR;
+}
+
+BitMask<WaterHeaterDemandBitmap> WhmManufacturer::DetermineHeatingSources()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmManufacturer()->GetWhmDelegate();
+    if (dg == nullptr)
+    {
+        ChipLogError(AppServer, "WhmDelegate is not initialized");
+        return BitMask<WaterHeaterDemandBitmap>(0);
+    }
+
+    // A list of valid heaterTypes
+    uint8_t waterHeaterTypeValues[] = {
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement1),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement2),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kHeatPump),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kBoiler),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kOther),
+    };
+
+    // The corresponding list of valid headerDemands
+    uint8_t waterHeaterDemandValues[] = {
+        static_cast<uint8_t>(WaterHeaterDemandBitmap::kImmersionElement1),
+        static_cast<uint8_t>(WaterHeaterDemandBitmap::kImmersionElement2),
+        static_cast<uint8_t>(WaterHeaterDemandBitmap::kHeatPump),
+        static_cast<uint8_t>(WaterHeaterDemandBitmap::kBoiler),
+        static_cast<uint8_t>(WaterHeaterDemandBitmap::kOther),
+    };
+
+    // Iterate across the valid waterHeaterTypes seeing which heating sources are available based on heaterTypes.
+    // Set the corresponding bit in the heaterDemand bitmap.
+    BitMask<WaterHeaterTypeBitmap> heaterTypes = dg->GetHeaterTypes();
+
+    uint8_t heaterDemandMask = 0;
+    for (uint16_t idx = 0; idx < static_cast<uint16_t>(sizeof(waterHeaterTypeValues) / sizeof(waterHeaterTypeValues[0])); idx++)
+    {
+        // Is this heating source being used?
+        if (heaterTypes.Raw() & waterHeaterTypeValues[idx])
```

Comment:
If waterHeaterTypeValues had the right type, and with the loop change to be clearer, this could be:
```suggestion
        if (heaterTypes.Has(waterHeaterTypeValue))
```

Reply:
Added this to https://github.com/project-chip/connectedhomeip/issues/34565 follow up issue.